packages/suite/src/components/wallet/CoinjoinSummary/CoinjoinSetupStrategies.tsx

```diff
@@ -1,94 +1,148 @@
+import { CoinjoinCustomStrategy } from './CoinJoinCustomStrategy';
+import { CoinjoinDefaultStrategy } from './CoinJoinDefaultStrategy';
+import { CryptoAmountWithHeader } from './CryptoAmountWithHeader';
 
-const Wrapper = styled(Card)`
-    width: 100%;
-    flex-direction: column;
+const StyledCard = styled(Card)`
+    margin-bottom: 8px;
 `;
 
-const Content = styled.div`
+const Row = styled.div`
     display: flex;
-    flex-direction: row;
-    padding-bottom: 24px;
+    gap: 50px;
 `;
 
-const Section = styled.div`
-    padding: 10px;
+const StyledCryptoAmountWithHeader = styled(CryptoAmountWithHeader)`
+    flex-basis: 50%;
 `;
 
-const SelectButton = styled(Button)<{ isActive?: boolean }>`
-    width: 200px;
-    height: 100px;
-    background: transparent;
-    border: 1px solid
-        ${props => (!props.isActive ? props.theme.BG_SECONDARY : props.theme.TYPE_GREEN)};
+const Heading = styled.div`
+    font-size: ${variables.FONT_SIZE.H3};
+    font-weight: ${variables.FONT_WEIGHT.MEDIUM};
 `;
 
-// Parameters sent to TrezorConnect.authorizeCoinjoin method
-const COINJOIN_STRATEGIES = {
-    default: {
-        maxRounds: 10,
-        anonymityLevel: 80,
-        maxFeePerKvbyte: 129000,
-        maxCoordinatorFeeRate: 0.003 * 10 ** 10, // 0.003 from coordinator
-    },
-    speed: {
-        maxRounds: 3,
-        anonymityLevel: 40,
-        maxFeePerKvbyte: 129000,
-        maxCoordinatorFeeRate: 0.003 * 10 ** 10, // 0.003 from coordinator
-    },
-};
+const ConfirmationsHeading = styled(Heading)`
+    margin-bottom: 4px;
+`;
+
+const AmountHeading = styled.div`
+    color: ${({ theme }) => theme.TYPE_GREEN};
+`;
+
+const Line = styled.hr`
+    border: none;
+    background: ${({ theme }) => theme.STROKE_GREY};
+    height: 1px;
+    margin: 20px 0 16px 0;
+`;
+
+const StyledCheckbox = styled(Checkbox)`
+    margin-top: 16px;
+`;
+
+const StyledButton = styled(Button)`
+    margin: 24px auto 0 auto;
+
+    :disabled {
+        background: ${({ theme }) => theme.STROKE_GREY};
+    }
+`;
 
 interface CoinjoinSetupStrategiesProps {
     account: Account;
 }
 
 export const CoinjoinSetupStrategies = ({ account }: CoinjoinSetupStrategiesProps) => {
-    const [strategy, setStrategy] = useState<keyof typeof COINJOIN_STRATEGIES>('default');
+    const [strategy, setStrategy] = useState<CoinJoinStrategy>('recommended');
+    const [connectedConfirmed, setConnectedConfirmed] = useState(false);
+    const [termsConfirmed, setTermsConfirmed] = useState(false);
+
     const actions = useActions({
         startCoinjoinSession,
     });
 
-    const selectedStrategy = COINJOIN_STRATEGIES[strategy];
+    const isCustom = strategy === 'custom';
+    const allChecked = connectedConfirmed && termsConfirmed;
+    const fee = (Number(account.balance) * 0.003).toString();
+
+    const reset = () => setStrategy('recommended');
+    const toggleConnectConfirmation = () => setConnectedConfirmed(!connectedConfirmed);
```

Comment:
For toggling it's always better to get used to 

```suggestion
    const toggleConnectConfirmation = () => setConnectedConfirmed(current => !current);
```

Reply:
3770dcd2737b99b21f4c2d835c606a32271e4214